### DISCUSSION TITLE
Is it possible to modify dims of graph nodes?

### DISCUSSION THREAD

**Discussion by CHNtentes** [onnx/onnx#6263] [Q&A]
I have an onnx model of llama2, and its input and output are static shape, with 1024 sequence length. Now I'd like to make a 256 sequence length model, and I know I can convert from pytorch again. I'm just curious whether it's possible to directly modify the dims inside onnx graph?

I tried changing the dims of model.graph.input and model.graph.output, but for all the nodes in between, I'm not sure how to do this...

**Comment by gramalingam** [comment]
It can be challenging to do this in general. I think what is easier and preferable is to export a model with the sequence length being dynamic (that is, a symbolic dimension called SeqLen or whatever). Once you have such a model, it is much easier to specialize it to a given length, by replacing SeqLen by 256 through out the model.
